server/stycoon/game.go

```diff
@@ -2,20 +2,250 @@ package stycoon
+		if err != nil {
+			return ships, fmt.Errorf("row read failed %v", err)
+		}
+		ship.Position = pos
+		ship.PrevPosition = posPrev
+		ship.Resources = map[string]Resource{}
+		err = game.setResources(id, &ship.Resources)
+		if err != nil {
+			return ships, err
+		}
+		if playerId != nil && *playerId == ship.Player {
+			ship.Command, err = game.getCommand(id)
+			if err != nil {
+				return ships, err
+			}
+		}
+		ships[strconv.Itoa(id)] = ship
+	}
+	if err = rows.Err(); err != nil {
+		return ships, fmt.Errorf("rows read failed: %v", err)
+	}
+	return ships, nil
+}
+
+func (game *Game) getCommand(shipId int) (Command, error) {
+	var command Command
+	err := game.db.QueryRow("select `type`, `target`, `resource`, `amount`, `class` from t_command where ship = ?", shipId).Scan(&command.Type, &command.Target, &command.Resource, &command.Amount, &command.ShipClass)
+	if err != nil {
+		return command, err
+	}
+	return command, nil
+}
+
+func (game *Game) setPlanetResourcePrices(planetId int, planetResources *map[string]TradingResource) error {
+	rows, err := game.db.Query("select `resource`, `buy`, `sell`  from t_price where planet = ?", planetId)
+	if err != nil {
+		return fmt.Errorf("query failed %v", err)
+	}
+	var resourceId int
+	var buy, sell sql.NullFloat64
+	for rows.Next() {
+		err = rows.Scan(&resourceId, &buy, &sell)
+		if err != nil {
+			return fmt.Errorf("row read failed %v", err)
+		}
+		if buy.Valid {
+			resource, ok := (*planetResources)[strconv.Itoa(resourceId)]
+			if ok {
+				(*planetResources)[strconv.Itoa(resourceId)] = TradingResource{Amount: resource.Amount, BuyPrice: buy.Float64}
+			} else {
+				(*planetResources)[strconv.Itoa(resourceId)] = TradingResource{BuyPrice: buy.Float64}
+			}
+		} else if sell.Valid {
+			resource, ok := (*planetResources)[strconv.Itoa(resourceId)]
+			if ok {
+				(*planetResources)[strconv.Itoa(resourceId)] = TradingResource{Amount: resource.Amount, SellPrice: sell.Float64}
+			} else {
+				(*planetResources)[strconv.Itoa(resourceId)] = TradingResource{SellPrice: sell.Float64}
+			}
+		}
+	}
+	if err = rows.Err(); err != nil {
+		return fmt.Errorf("rows read failed: %v", err)
+	}
+	return nil
+}
+
+func (game *Game) setTradingResources(objectId int, resources *map[string]TradingResource) error {
+	amounts, err := game.getCommodityAmounts(objectId)
+	if err != nil {
+		return nil
+	}
+	for key, amount := range amounts {
+		(*resources)[key] = TradingResource{Amount: amount}
+	}
+	return nil
+}
+
+func (game *Game) setResources(objectId int, resources *map[string]Resource) error {
+	amounts, err := game.getCommodityAmounts(objectId)
+	if err != nil {
+		return nil
+	}
+	for key, amount := range amounts {
+		(*resources)[key] = Resource{Amount: amount}
+	}
+	return nil
+}
+
+func (game *Game) getCommodityAmounts(objectId int) (map[string]int64, error) {
+	amounts := make(map[string]int64)
+	rows, err := game.db.Query("select `resource`, `amount` from t_commodity where object = ?", objectId)
```

Comment:
again. one command for each planed and ship. kills performance.

Reply:
optimized